src/documents/barcodes.py

```diff
@@ -248,11 +276,17 @@ def detect(self) -> None:
                         (int(round(x * factor)), (int(round(y * factor)))),
                     )
 
+                # Detect barcodes
+                logger.debug("Detecting barcodes")
                 for barcode_value in reader(page):
                     self.barcodes.append(
                         Barcode(current_page_number, barcode_value),
                     )
 
+                # Delete temporary image file
+                logger.debug("Cleaning up temporary image file")
+                os.remove(page_filename)
```

Comment:
```suggestion
                Path(page_filename).unlink()
```

Should double-check this works but in general I think this is preferred

Reply:
From what I understand from the documentation both ways should work exactly the same. https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

The `Path().unlink()` has the advantage that it can work directly on Path objects and does not need the path to be handled as string. Since we have the path as string from the pdf2image package this advantage does not apply. In theory creating a path object which gets converted back to a string for the passing to a syscall could be a small overhead but I think it is negligible. We should choose here what is used in the paperless codebase. If you think that you suggestion would fit better I am also happy to commit it.

Reply:
We try to use Path everywhere

Reply:
I rewrote that a little and converted the path to a pathlib.Path object as early as possible to be more consistent with usage of Path where possible.